### PR TITLE
Fixes #7533 - center pie chart labels and prevent overflow

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

**Comment by domcleal** [comment]
It might be preferable to reduce the font size, or even auto-fit it if that's possible, rather than introducing a line wrap?

**Comment by tbrisker** [comment]
Reducing font size might work for latin languages, but asian (such a ja_JP) characters would need to be reduced to a very small size (<12px) for some strings.
Autofitting is possible, but would require more work than I think is worth investing here, and would still cause some asain strings to be very small.
Would you say that a very small font is better then line wrap here?

**Comment by domcleal** [comment]
I don't know, not my area of expertise... but I'm aware that although the browser is responsible for line wrapping in an appropriate place, it often doesn't do well with CJK style languages as we've had other bug reports for it.  I'd half expect to be told again after merging this that it shouldn't then wrap.

**Comment by tbrisker** [comment]
@domcleal added auto-resize on overflow, as the translator said it would be best to resize. This now only resizes labels that will overflow, to prevent small fonts when not necessary.

File: app/assets/javascripts/charts.js [hunk 1]
```diff
@@ -44,9 +44,19 @@ $.fn.flot_pie = function(){
     $(target).bind("plotclick", function (event, pos, item) {
       search_on_click(event, item);
     });
+    resize_label(target.children('.pieLabel').first());
   });
 };
 
+function resize_label(label){
+    var labelOffset = parseInt(label.css('left'));
```
Comment (domcleal) [inline app/assets/javascripts/charts.js:52]: Nitpick, two space indentation to match the rest of the file.

**Comment by domcleal** [comment]
Merged as 5d0a54a, thanks @tbrisker.

**Pull request closed without merging**